util/bloom/xor.go

```diff
@@ -93,28 +94,37 @@ func (xf *XorFilter) MarshalBinary() ([]byte, error) {
 // ErrBadBinary is returned when UnmarshalBinary fails
 var ErrBadBinary = errors.New("bad XorFilter binary")
 
+// TODO: make this an option to UnmarshalBinary, or a settable global, ...
+const maxFingerprints = 1000000
```

Comment:
I agree with the TODO. We can do that later on. Sizing also seems to be reasonable for what we need.